packages/tools/src/errors.ts

```diff
@@ -0,0 +1,32 @@
+export class RuntimeError extends Error {
+  public name = 'RuntimeError'
+  public originalError?: Error
+
+  public constructor(error: Error)
+  public constructor(message: string)
+
+  public constructor(...params: unknown[]) {
+    if (params.length === 2) {
+      const message: string = (
+        typeof params[0] === 'string' ? params[0] : params[1]
+      ) as string
+
+      const originalError: Error = (
+        typeof params[0] === 'string' ? params[1] : params[0]
+      ) as Error
```

Comment:
add signature type with 2 arguments, then you won't need to do those ternary operations

```
public constructor(errorOrMessage: string | Error)
public constructor(message: string, error: Error)
public constructor(errorOrMessage: string | Error, error?: Error)
```